var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -245,256 +242,173 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
+            intsize = "64" if '+int64' in spec else "32"
+            args.extend([
+                define('SUNDIALS_INDEX_SIZE', intsize),
+                define('SUNDIALS_INDEX_TYPE', 'int{}_t'.format(intsize)),
+            ])
 
-        # Monitoring
         args.extend([
-            self.define_from_variant('SUNDIALS_BUILD_WITH_MONITORING', 'monitoring')
+            # Fortran interface
+            from_variant('F77_INTERFACE_ENABLE', 'fcmix'),
+            from_variant('F2003_INTERFACE_ENABLE', 'f2003'),
+            # library type
+            from_variant('BUILD_SHARED_LIBS', 'shared'),
+            from_variant('BUILD_STATIC_LIBS', 'static'),
+            # Generic (std-c) math libraries
+            from_variant('USE_GENERIC_MATH', 'generic-math'),
+            # Monitoring
+            from_variant('SUNDIALS_BUILD_WITH_MONITORING', 'monitoring'),
+            # Profiling
+            from_variant('SUNDIALS_BUILD_WITH_PROFILING', 'profiling'),
+            from_variant('ENABLE_CALIPER', 'caliper'),
         ])
 
-        # Profiling
-        args.extend([
-            self.define_from_variant('SUNDIALS_BUILD_WITH_PROFILING', 'profiling')
-        ])
-        if '+profiling+caliper' in spec:
-            args.extend([
-                '-DENABLE_CALIPER=ON',
-                '-DCALIPER_DIR=%s' % spec['caliper'].prefix
-            ])
+        if '+caliper' in spec:
+            args.append(define('CALIPER_DIR', spec['caliper'].prefix))
 
         # parallelism
         args.extend([
-            self.define_from_variant('MPI_ENABLE', 'mpi'),
-            self.define_from_variant('OPENMP_ENABLE', 'openmp'),
-            self.define_from_variant('PTHREAD_ENABLE', 'pthread'),
-            self.define_from_variant('ENABLE_SYCL', 'sycl')
+            from_variant('MPI_ENABLE', 'mpi'),
+            from_variant('OPENMP_ENABLE', 'openmp'),
+            from_variant('PTHREAD_ENABLE', 'pthread'),
+            from_variant('ENABLE_SYCL', 'sycl'),
+            from_variant('CUDA_ENABLE', 'cuda'),
+            from_variant('ENABLE_HIP', 'rocm'),
+            from_variant('HYPRE_ENABLE', 'hypre'),
+            from_variant('KLU_ENABLE', 'klu'),
+            from_variant('LAPACK_ENABLE', 'lapack'),
+            from_variant('PETSC_ENABLE', 'petsc'),
+            from_variant('RAJA_ENABLE', 'raja'),
+            from_variant('SUPERLUMT_ENABLE', 'superlu-mt'),
+            from_variant('SUPERLUDIST_ENABLE', 'superlu-dist'),
+            from_variant('Trilinos_ENABLE', 'trilinos'),
+            from_variant('EXAMPLES_INSTALL', 'examples-install'),
         ])
 
         if '+cuda' in spec:
-            args.append('-DCUDA_ENABLE=ON')
-            archs = spec.variants['cuda_arch'].value
-            if archs[0] != 'none':
-                arch_str = ",".join(archs)
-                args.append('-DCMAKE_CUDA_ARCHITECTURES=%s' % arch_str)
-        else:
-            args.append('-DCUDA_ENABLE=OFF')
+            args.append(define(
```

Comment:
I assume `define` is smart enough to handle `spec.variants['cuda_arch'].value` being a list, is that right?

Reply:
Correct! `define` and `from_variant` will turn a list into a CMake-separated semicolon list, which is what `CMAKE_CUDA_ARCHITECTURES` needs. (The previous recipe had `,` which I'm not sure is valid—and might not be tested because most people install with only a single arch.)

Reply:
Cool. I'll test out the changes, but they look good to me.